lib/hierarchyShape.ts

```diff
@@ -0,0 +1,71 @@
+import clownface, { GraphPointer } from 'clownface'
+import { rdf, sh, dashSparql } from '@tpluscode/rdf-ns-builders'
+import { meta } from '@zazuko/vocabulary-extras/builders'
+import { isGraphPointer, isNamedNode } from 'is-graph-pointer'
+import { s2q } from '@hydrofoil/shape-to-query'
+
+export function fromHierarchy(hierarchy: GraphPointer) {
+  const clone = clownface({
+    dataset: hierarchy.dataset.match(),
+    term: hierarchy.term,
+  })
+
+  const rootShape = clone.blankNode()
+
+  rootShape
+    .addOut(rdf.type, sh.NodeShape)
+    .addOut(sh.targetNode, hierarchy.out(meta.hierarchyRoot))
+    .addOut(sh.rule, rule => {
+      rule.addOut(rdf.type, s2q.SPORule)
```

Comment:
To mimic `DESCRIBE`, I add this [SPO Rule](https://shape-to-query.hypermedia.app/docs/#extension-spo-rule) which generates open-ended `?s ?p ?o` patterns to every hierarchy level